failed: ClusterShardingRememberEntitiesSpec: Cluster sharding with remember entities (persistence) must setup shared journal

https://jenkins.akka.io:8498/job/akka-artery-cluster-tests/2084/consoleFull
```
[info] [JVM-1] - must setup shared journal (on node 'first', class akka.cluster.sharding.PersistentClusterShardingRememberEntitiesMultiJvmNode1) *** FAILED ***
[info] [JVM-1]   java.lang.AssertionError: assertion failed: timeout (10 seconds) during expectMsgClass waiting for class akka.actor.ActorIdentity
[info] [JVM-1]   at scala.Predef$.assert(Predef.scala:223)
[info] [JVM-1]   at akka.testkit.TestKitBase.expectMsgClass_internal(TestKit.scala:528)
[info] [JVM-1]   at akka.testkit.TestKitBase.expectMsgType(TestKit.scala:510)
[info] [JVM-1]   at akka.testkit.TestKitBase.expectMsgType$(TestKit.scala:509)
[info] [JVM-1]   at akka.testkit.TestKit.expectMsgType(TestKit.scala:928)
[info] [JVM-1]   at akka.cluster.sharding.ClusterShardingRememberEntitiesSpec.$anonfun$new$4(ClusterShardingRememberEntitiesSpec.scala:163)
[info] [JVM-1]   at akka.remote.testkit.MultiNodeSpec.runOn(MultiNodeSpec.scala:384)
[info] [JVM-1]   at akka.cluster.sharding.ClusterShardingRememberEntitiesSpec.$anonfun$new$2(ClusterShardingRememberEntitiesSpec.scala:161)
[info] [JVM-1]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info] [JVM-1]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
```

--- a/akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingRememberEntitiesSpec.scala
+++ b/akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingRememberEntitiesSpec.scala
@@ -95,46 +95,40 @@ abstract class ClusterShardingRememberEntitiesSpecConfig(val mode: String, val r
 
 }
 
-object PersistentClusterShardingRememberEntitiesEnabledSpecConfig
-    extends ClusterShardingRememberEntitiesSpecConfig(ClusterShardingSettings.StateStoreModePersistence, true)
-object PersistentClusterShardingRememberEntitiesDefaultSpecConfig
-    extends ClusterShardingRememberEntitiesSpecConfig(ClusterShardingSettings.StateStoreModePersistence, false)
-object DDataClusterShardingRememberEntitiesEnabledSpecConfig
-    extends ClusterShardingRememberEntitiesSpecConfig(ClusterShardingSettings.StateStoreModeDData, true)
-object DDataClusterShardingRememberEntitiesDefaultSpecConfig
-    extends ClusterShardingRememberEntitiesSpecConfig(ClusterShardingSettings.StateStoreModeDData, false)
+class PersistentClusterShardingRememberEntitiesSpecConfig(rememberEntities: Boolean)
+    extends ClusterShardingRememberEntitiesSpecConfig(
+      ClusterShardingSettings.StateStoreModePersistence,
+      rememberEntities)
+class DDataClusterShardingRememberEntitiesSpecConfig(rememberEntities: Boolean)
+    extends ClusterShardingRememberEntitiesSpecConfig(ClusterShardingSettings.StateStoreModeDData, rememberEntities)
 
-class PersistentClusterShardingRememberEntitiesEnabledSpec
-    extends ClusterShardingRememberEntitiesSpec(PersistentClusterShardingRememberEntitiesEnabledSpecConfig)
-class PersistentClusterShardingRememberEntitiesDefaultSpec
-    extends ClusterShardingRememberEntitiesSpec(PersistentClusterShardingRememberEntitiesDefaultSpecConfig)
+abstract class PersistentClusterShardingRememberEntitiesSpec(val rememberEntities: Boolean)
+    extends ClusterShardingRememberEntitiesSpec(
+      new PersistentClusterShardingRememberEntitiesSpecConfig(rememberEntities))
+abstract class DDataClusterShardingRememberEntitiesSpec(val rememberEntities: Boolean)
+    extends ClusterShardingRememberEntitiesSpec(new DDataClusterShardingRememberEntitiesSpecConfig(rememberEntities))
 
 class PersistentClusterShardingRememberEntitiesEnabledMultiJvmNode1
-    extends PersistentClusterShardingRememberEntitiesEnabledSpec
+    extends PersistentClusterShardingRememberEntitiesSpec(true)
 class PersistentClusterShardingRememberEntitiesEnabledMultiJvmNode2
-    extends PersistentClusterShardingRememberEntitiesEnabledSpec
+    extends PersistentClusterShardingRememberEntitiesSpec(true)
 class PersistentClusterShardingRememberEntitiesEnabledMultiJvmNode3
-    extends PersistentClusterShardingRememberEntitiesEnabledSpec
+    extends PersistentClusterShardingRememberEntitiesSpec(true)
 
 class PersistentClusterShardingRememberEntitiesDefaultMultiJvmNode1
-    extends PersistentClusterShardingRememberEntitiesDefaultSpec
+    extends PersistentClusterShardingRememberEntitiesSpec(false)
 class PersistentClusterShardingRememberEntitiesDefaultMultiJvmNode2
-    extends PersistentClusterShardingRememberEntitiesDefaultSpec
+    extends PersistentClusterShardingRememberEntitiesSpec(false)
 class PersistentClusterShardingRememberEntitiesDefaultMultiJvmNode3
-    extends PersistentClusterShardingRememberEntitiesDefaultSpec
+    extends PersistentClusterShardingRememberEntitiesSpec(false)
 
-class DDataClusterShardingRememberEntitiesEnabledSpec
-    extends ClusterShardingRememberEntitiesSpec(DDataClusterShardingRememberEntitiesEnabledSpecConfig)
-class DDataClusterShardingRememberEntitiesDefaultSpec
-    extends ClusterShardingRememberEntitiesSpec(DDataClusterShardingRememberEntitiesEnabledSpecConfig)
+class DDataClusterShardingRememberEntitiesEnabledMultiJvmNode1 extends DDataClusterShardingRememberEntitiesSpec(true)
+class DDataClusterShardingRememberEntitiesEnabledMultiJvmNode2 extends DDataClusterShardingRememberEntitiesSpec(true)
+class DDataClusterShardingRememberEntitiesEnabledMultiJvmNode3 extends DDataClusterShardingRememberEntitiesSpec(true)
 
-class DDataClusterShardingRememberEntitiesEnabledMultiJvmNode1 extends DDataClusterShardingRememberEntitiesEnabledSpec
-class DDataClusterShardingRememberEntitiesEnabledMultiJvmNode2 extends DDataClusterShardingRememberEntitiesEnabledSpec
-class DDataClusterShardingRememberEntitiesEnabledMultiJvmNode3 extends DDataClusterShardingRememberEntitiesEnabledSpec
-
-class DDataClusterShardingRememberEntitiesDisabledMultiJvmNode1 extends DDataClusterShardingRememberEntitiesEnabledSpec
-class DDataClusterShardingRememberEntitiesDisabledMultiJvmNode2 extends DDataClusterShardingRememberEntitiesEnabledSpec
-class DDataClusterShardingRememberEntitiesDisabledMultiJvmNode3 extends DDataClusterShardingRememberEntitiesEnabledSpec
+class DDataClusterShardingRememberEntitiesDefaultMultiJvmNode1 extends DDataClusterShardingRememberEntitiesSpec(false)
+class DDataClusterShardingRememberEntitiesDefaultMultiJvmNode2 extends DDataClusterShardingRememberEntitiesSpec(false)
+class DDataClusterShardingRememberEntitiesDefaultMultiJvmNode3 extends DDataClusterShardingRememberEntitiesSpec(false)
 
 abstract class ClusterShardingRememberEntitiesSpec(config: ClusterShardingRememberEntitiesSpecConfig)
     extends MultiNodeSpec(config)
@@ -184,6 +178,14 @@ abstract class ClusterShardingRememberEntitiesSpec(config: ClusterShardingRememb
 
   def isDdataMode: Boolean = mode == ClusterShardingSettings.StateStoreModeDData
 
+  def expectEntityRestarted(sys: ActorSystem, event: Int, probe: TestProbe, entityProbe: TestProbe): Started =
+    if (!rememberEntities) {
+      probe.send(ClusterSharding(sys).shardRegion(dataType), event)
+      probe.expectMsg(1)
+      entityProbe.expectMsgType[Started](20.seconds) // passes
+    } else
+      entityProbe.expectMsgType[Started](20.seconds) // FIXME fails if enabled and persistence
+
   def setStoreIfNotDdata(sys: ActorSystem): Unit =
     if (!isDdataMode) {
       val probe = TestProbe()(sys)
@@ -212,17 +214,20 @@ abstract class ClusterShardingRememberEntitiesSpec(config: ClusterShardingRememb
     }
 
     "start remembered entities when coordinator fail over" in within(30.seconds) {
+      val entityProbe = TestProbe()
+      val probe = TestProbe()
       join(second, second)
       runOn(second) {
-        startSharding(system, testActor)
-        region ! 1
-        expectMsgType[Started]
+        startSharding(system, entityProbe.ref)
+        probe.send(region, 1)
+        probe.expectMsg(1)
+        entityProbe.expectMsgType[Started]
       }
       enterBarrier("second-started")
 
       join(third, second)
       runOn(third) {
-        startSharding(system, testActor)
+        startSharding(system, entityProbe.ref)
       }
       runOn(second, third) {
         within(remaining) {
@@ -246,21 +251,7 @@ abstract class ClusterShardingRememberEntitiesSpec(config: ClusterShardingRememb
       enterBarrier("crash-second")
 
       runOn(third) {
-        if (!rememberEntities) {
-          ClusterSharding(system).shardRegion(dataType) ! 1
-        }
-        // FIXME This is where we fail if remember=true, mode=persistence.
-        // What is curious is after node 'second' stops
-        // the ShardCoordinator does correctly restart on node 'third',
-        // however the entity does not get re-created automatically as it should, even though:
-        // the ShardRegion
-        // - does get registered: [Actor[akka://ClusterShardingRememberEntitiesSpec/system/sharding/Entity#380407716]]
-        // - is in the ShardCoordinator's 'aliveRegion'
-        // - does receive the RegisterAck
-        expectMsgType[Started](20.seconds)
-        if (!rememberEntities) {
-          expectMsg(1)
-        }
+        expectEntityRestarted(system, 1, probe, entityProbe)
       }
 
       enterBarrier("after-2")
@@ -277,20 +268,16 @@ abstract class ClusterShardingRememberEntitiesSpec(config: ClusterShardingRememb
         // no nodes left of the original cluster, start a new cluster
 
         val sys2 = ActorSystem(system.name, system.settings.config)
+        val entityProbe2 = TestProbe()(sys2)
         val probe2 = TestProbe()(sys2)
 
         setStoreIfNotDdata(sys2)
 
         Cluster(sys2).join(Cluster(sys2).selfAddress)
-        startSharding(sys2, probe2.ref)
 
-        if (!rememberEntities) {
-          probe2.send(ClusterSharding(sys2).shardRegion(dataType), 1)
-        }
-        probe2.expectMsgType[Started](20.seconds)
-        if (!rememberEntities) {
-          probe2.expectMsg(1)
-        }
+        startSharding(sys2, entityProbe2.ref)
+
+        expectEntityRestarted(sys2, 1, probe2, entityProbe2)
 
         shutdown(sys2)
       }

--- a/akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingRememberEntitiesSpec.scala
+++ b/akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingRememberEntitiesSpec.scala
@@ -46,63 +46,95 @@ object ClusterShardingRememberEntitiesSpec {
 
 }
 
-abstract class ClusterShardingRememberEntitiesSpecConfig(val mode: String) extends MultiNodeConfig {
+abstract class ClusterShardingRememberEntitiesSpecConfig(val mode: String, val rememberEntities: Boolean)
+    extends MultiNodeConfig {
   val first = role("first")
   val second = role("second")
   val third = role("third")
 
+  val targetDir = s"target/ClusterShardingRememberEntitiesSpec-$mode-remember-$rememberEntities"
+
+  val modeConfig =
+    if (mode == ClusterShardingSettings.StateStoreModeDData) ConfigFactory.empty
+    else ConfigFactory.parseString(s"""
+      akka.persistence.journal.plugin = "akka.persistence.journal.leveldb-shared"
+      akka.persistence.journal.leveldb-shared.timeout = 5s
+      akka.persistence.journal.leveldb-shared.store.native = off
+      akka.persistence.journal.leveldb-shared.store.dir = "$targetDir/journal"
+      akka.persistence.snapshot-store.plugin = "akka.persistence.snapshot-store.local"
+      akka.persistence.snapshot-store.local.dir = "$targetDir/snapshots"
+      akka.cluster.sharding.journal-plugin-id = "akka.persistence.journal.inmem"
+      """)
+
   commonConfig(
-    ConfigFactory
-      .parseString(s"""
-    akka.loglevel = DEBUG
-    akka.actor.provider = "cluster"
-    akka.cluster.auto-down-unreachable-after = 0s
-    akka.remote.log-remote-lifecycle-events = off
-    akka.persistence.journal.plugin = "akka.persistence.journal.leveldb-shared"
-    akka.persistence.journal.leveldb-shared {
-      timeout = 5s
-      store {
-        native = off
-        dir = "target/ShardingRememberEntitiesSpec/journal"
+    modeConfig
+      .withFallback(ConfigFactory.parseString(s"""
+      akka.loglevel = DEBUG
+      akka.actor.provider = "cluster"
+      akka.cluster.auto-down-unreachable-after = 0s
+      akka.remote.log-remote-lifecycle-events = off
+      akka.cluster.sharding.state-store-mode = "$mode"
+      akka.cluster.sharding.distributed-data.durable.lmdb {
+        dir = $targetDir/sharding-ddata
+        map-size = 10 MiB
       }
-    }
-    akka.persistence.snapshot-store.plugin = "akka.persistence.snapshot-store.local"
-    akka.persistence.snapshot-store.local.dir = "target/ShardingRememberEntitiesSpec/snapshots"
-    akka.cluster.sharding.state-store-mode = "$mode"
-    akka.cluster.sharding.distributed-data.durable.lmdb {
-      dir = target/ShardingRememberEntitiesSpec/sharding-ddata
-      map-size = 10 MiB
-    }
-    """)
+      # FIXME temporary while sanity-checking persistence enabled
+      # waiting for `Started` from restarted entity
+      akka.testconductor.barrier-timeout = 60 s
+      akka.test.single-expect-default = 60 s # defaults to 3s
+      """))
       .withFallback(SharedLeveldbJournal.configToEnableJavaSerializationForTest)
       .withFallback(MultiNodeClusterSpec.clusterConfig))
 
   nodeConfig(third)(ConfigFactory.parseString(s"""
     akka.cluster.sharding.distributed-data.durable.lmdb {
       # use same directory when starting new node on third (not used at same time)
-      dir = target/ShardingRememberEntitiesSpec/sharding-third
+      dir = $targetDir/sharding-third
     }
     """))
+
 }
 
-object PersistentClusterShardingRememberEntitiesSpecConfig
-    extends ClusterShardingRememberEntitiesSpecConfig(ClusterShardingSettings.StateStoreModePersistence)
-object DDataClusterShardingRememberEntitiesSpecConfig
-    extends ClusterShardingRememberEntitiesSpecConfig(ClusterShardingSettings.StateStoreModeDData)
+object PersistentClusterShardingRememberEntitiesEnabledSpecConfig
+    extends ClusterShardingRememberEntitiesSpecConfig(ClusterShardingSettings.StateStoreModePersistence, true)
+object PersistentClusterShardingRememberEntitiesDefaultSpecConfig
+    extends ClusterShardingRememberEntitiesSpecConfig(ClusterShardingSettings.StateStoreModePersistence, false)
+object DDataClusterShardingRememberEntitiesEnabledSpecConfig
+    extends ClusterShardingRememberEntitiesSpecConfig(ClusterShardingSettings.StateStoreModeDData, true)
+object DDataClusterShardingRememberEntitiesDefaultSpecConfig
+    extends ClusterShardingRememberEntitiesSpecConfig(ClusterShardingSettings.StateStoreModeDData, false)
 
-class PersistentClusterShardingRememberEntitiesSpec
-    extends ClusterShardingRememberEntitiesSpec(PersistentClusterShardingRememberEntitiesSpecConfig)
+class PersistentClusterShardingRememberEntitiesEnabledSpec
+    extends ClusterShardingRememberEntitiesSpec(PersistentClusterShardingRememberEntitiesEnabledSpecConfig)
+class PersistentClusterShardingRememberEntitiesDefaultSpec
+    extends ClusterShardingRememberEntitiesSpec(PersistentClusterShardingRememberEntitiesDefaultSpecConfig)
 
-class PersistentClusterShardingRememberEntitiesMultiJvmNode1 extends PersistentClusterShardingRememberEntitiesSpec
-class PersistentClusterShardingRememberEntitiesMultiJvmNode2 extends PersistentClusterShardingRememberEntitiesSpec
-class PersistentClusterShardingRememberEntitiesMultiJvmNode3 extends PersistentClusterShardingRememberEntitiesSpec
+class PersistentClusterShardingRememberEntitiesEnabledMultiJvmNode1
+    extends PersistentClusterShardingRememberEntitiesEnabledSpec
+class PersistentClusterShardingRememberEntitiesEnabledMultiJvmNode2
+    extends PersistentClusterShardingRememberEntitiesEnabledSpec
+class PersistentClusterShardingRememberEntitiesEnabledMultiJvmNode3
+    extends PersistentClusterShardingRememberEntitiesEnabledSpec
 
-class DDataClusterShardingRememberEntitiesSpec
-    extends ClusterShardingRememberEntitiesSpec(DDataClusterShardingRememberEntitiesSpecConfig)
+class PersistentClusterShardingRememberEntitiesDefaultMultiJvmNode1
+    extends PersistentClusterShardingRememberEntitiesDefaultSpec
+class PersistentClusterShardingRememberEntitiesDefaultMultiJvmNode2
+    extends PersistentClusterShardingRememberEntitiesDefaultSpec
+class PersistentClusterShardingRememberEntitiesDefaultMultiJvmNode3
+    extends PersistentClusterShardingRememberEntitiesDefaultSpec
 
-class DDataClusterShardingRememberEntitiesMultiJvmNode1 extends DDataClusterShardingRememberEntitiesSpec
-class DDataClusterShardingRememberEntitiesMultiJvmNode2 extends DDataClusterShardingRememberEntitiesSpec
-class DDataClusterShardingRememberEntitiesMultiJvmNode3 extends DDataClusterShardingRememberEntitiesSpec
+class DDataClusterShardingRememberEntitiesEnabledSpec
+    extends ClusterShardingRememberEntitiesSpec(DDataClusterShardingRememberEntitiesEnabledSpecConfig)
+class DDataClusterShardingRememberEntitiesDefaultSpec
+    extends ClusterShardingRememberEntitiesSpec(DDataClusterShardingRememberEntitiesEnabledSpecConfig)
+
+class DDataClusterShardingRememberEntitiesEnabledMultiJvmNode1 extends DDataClusterShardingRememberEntitiesEnabledSpec
+class DDataClusterShardingRememberEntitiesEnabledMultiJvmNode2 extends DDataClusterShardingRememberEntitiesEnabledSpec
+class DDataClusterShardingRememberEntitiesEnabledMultiJvmNode3 extends DDataClusterShardingRememberEntitiesEnabledSpec
+
+class DDataClusterShardingRememberEntitiesDisabledMultiJvmNode1 extends DDataClusterShardingRememberEntitiesEnabledSpec
+class DDataClusterShardingRememberEntitiesDisabledMultiJvmNode2 extends DDataClusterShardingRememberEntitiesEnabledSpec
+class DDataClusterShardingRememberEntitiesDisabledMultiJvmNode3 extends DDataClusterShardingRememberEntitiesEnabledSpec
 
 abstract class ClusterShardingRememberEntitiesSpec(config: ClusterShardingRememberEntitiesSpecConfig)
     extends MultiNodeSpec(config)
@@ -111,7 +143,9 @@ abstract class ClusterShardingRememberEntitiesSpec(config: ClusterShardingRememb
   import ClusterShardingRememberEntitiesSpec._
   import config._
 
-  override def initialParticipants = roles.size
+  override def initialParticipants: Int = roles.size
+
+  val dataType = "Entity"
 
   val storageLocations = List(
     new File(system.settings.config.getString("akka.cluster.sharding.distributed-data.durable.lmdb.dir")).getParentFile)
@@ -128,24 +162,35 @@ abstract class ClusterShardingRememberEntitiesSpec(config: ClusterShardingRememb
   def join(from: RoleName, to: RoleName): Unit = {
     runOn(from) {
       Cluster(system).join(node(to).address)
+      awaitAssert {
+        Cluster(system).state.isMemberUp(node(from).address)
+      }
     }
     enterBarrier(from.name + "-joined")
   }
 
   val cluster = Cluster(system)
 
-  def startSharding(sys: ActorSystem = system, probe: ActorRef = testActor): Unit = {
+  def startSharding(sys: ActorSystem, probe: ActorRef): ActorRef = {
     ClusterSharding(sys).start(
-      typeName = "Entity",
+      typeName = dataType,
       entityProps = ClusterShardingRememberEntitiesSpec.props(probe),
-      settings = ClusterShardingSettings(system).withRememberEntities(true),
+      settings = ClusterShardingSettings(sys).withRememberEntities(rememberEntities),
       extractEntityId = extractEntityId,
       extractShardId = extractShardId)
   }
 
-  lazy val region = ClusterSharding(system).shardRegion("Entity")
+  lazy val region = ClusterSharding(system).shardRegion(dataType)
 
   def isDdataMode: Boolean = mode == ClusterShardingSettings.StateStoreModeDData
+
+  def setStoreIfNotDdata(sys: ActorSystem): Unit =
+    if (!isDdataMode) {
+      val probe = TestProbe()(sys)
+      sys.actorSelection(node(first) / "user" / "store").tell(Identify(None), probe.ref)
+      val sharedStore = probe.expectMsgType[ActorIdentity](20.seconds).ref.get
+      SharedLeveldbJournal.setStore(sharedStore, sys)
+    }
 
   s"Cluster sharding with remember entities ($mode)" must {
 
@@ -156,12 +201,10 @@ abstract class ClusterShardingRememberEntitiesSpec(config: ClusterShardingRememb
         runOn(first) {
           system.actorOf(Props[SharedLeveldbStore], "store")
         }
-        enterBarrier("peristence-started")
+        enterBarrier("persistence-started")
 
         runOn(first, second, third) {
-          system.actorSelection(node(first) / "user" / "store") ! Identify(None)
-          val sharedStore = expectMsgType[ActorIdentity](10.seconds).ref.get
-          SharedLeveldbJournal.setStore(sharedStore, system)
+          setStoreIfNotDdata(system)
         }
 
         enterBarrier("after-1")
@@ -171,7 +214,7 @@ abstract class ClusterShardingRememberEntitiesSpec(config: ClusterShardingRememb
     "start remembered entities when coordinator fail over" in within(30.seconds) {
       join(second, second)
       runOn(second) {
-        startSharding()
+        startSharding(system, testActor)
         region ! 1
         expectMsgType[Started]
       }
@@ -179,7 +222,7 @@ abstract class ClusterShardingRememberEntitiesSpec(config: ClusterShardingRememb
 
       join(third, second)
       runOn(third) {
-        startSharding()
+        startSharding(system, testActor)
       }
       runOn(second, third) {
         within(remaining) {
@@ -203,7 +246,21 @@ abstract class ClusterShardingRememberEntitiesSpec(config: ClusterShardingRememb
       enterBarrier("crash-second")
 
       runOn(third) {
-        expectMsgType[Started](remaining)
+        if (!rememberEntities) {
+          ClusterSharding(system).shardRegion(dataType) ! 1
+        }
+        // FIXME This is where we fail if remember=true, mode=persistence.
+        // What is curious is after node 'second' stops
+        // the ShardCoordinator does correctly restart on node 'third',
+        // however the entity does not get re-created automatically as it should, even though:
+        // the ShardRegion
+        // - does get registered: [Actor[akka://ClusterShardingRememberEntitiesSpec/system/sharding/Entity#380407716]]
+        // - is in the ShardCoordinator's 'aliveRegion'
+        // - does receive the RegisterAck
+        expectMsgType[Started](20.seconds)
+        if (!rememberEntities) {
+          expectMsg(1)
+        }
       }
 
       enterBarrier("after-2")
@@ -222,15 +279,18 @@ abstract class ClusterShardingRememberEntitiesSpec(config: ClusterShardingRememb
         val sys2 = ActorSystem(system.name, system.settings.config)
         val probe2 = TestProbe()(sys2)
 
-        if (!isDdataMode) {
-          sys2.actorSelection(node(first) / "user" / "store").tell(Identify(None), probe2.ref)
-          val sharedStore = probe2.expectMsgType[ActorIdentity](10.seconds).ref.get
-          SharedLeveldbJournal.setStore(sharedStore, sys2)
-        }
+        setStoreIfNotDdata(sys2)
 
         Cluster(sys2).join(Cluster(sys2).selfAddress)
         startSharding(sys2, probe2.ref)
+
+        if (!rememberEntities) {
+          probe2.send(ClusterSharding(sys2).shardRegion(dataType), 1)
+        }
         probe2.expectMsgType[Started](20.seconds)
+        if (!rememberEntities) {
+          probe2.expectMsg(1)
+        }
 
         shutdown(sys2)
       }

--- a/akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingRememberEntitiesSpec.scala
+++ b/akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingRememberEntitiesSpec.scala
@@ -63,7 +63,6 @@ abstract class ClusterShardingRememberEntitiesSpecConfig(val mode: String, val r
       akka.persistence.journal.leveldb-shared.store.dir = "$targetDir/journal"
       akka.persistence.snapshot-store.plugin = "akka.persistence.snapshot-store.local"
       akka.persistence.snapshot-store.local.dir = "$targetDir/snapshots"
-      akka.cluster.sharding.journal-plugin-id = "akka.persistence.journal.inmem"
       """)
 
   commonConfig(
@@ -78,10 +77,8 @@ abstract class ClusterShardingRememberEntitiesSpecConfig(val mode: String, val r
         dir = $targetDir/sharding-ddata
         map-size = 10 MiB
       }
-      # FIXME temporary while sanity-checking persistence enabled
-      # waiting for `Started` from restarted entity
       akka.testconductor.barrier-timeout = 60 s
-      akka.test.single-expect-default = 60 s # defaults to 3s
+      akka.test.single-expect-default = 60 s
       """))
       .withFallback(SharedLeveldbJournal.configToEnableJavaSerializationForTest)
       .withFallback(MultiNodeClusterSpec.clusterConfig))
@@ -178,13 +175,14 @@ abstract class ClusterShardingRememberEntitiesSpec(config: ClusterShardingRememb
 
   def isDdataMode: Boolean = mode == ClusterShardingSettings.StateStoreModeDData
 
-  def expectEntityRestarted(sys: ActorSystem, event: Int, probe: TestProbe, entityProbe: TestProbe): Started =
+  def expectEntityRestarted(sys: ActorSystem, event: Int, probe: TestProbe, entityProbe: TestProbe): Started = {
     if (!rememberEntities) {
       probe.send(ClusterSharding(sys).shardRegion(dataType), event)
       probe.expectMsg(1)
-      entityProbe.expectMsgType[Started](20.seconds) // passes
-    } else
-      entityProbe.expectMsgType[Started](20.seconds) // FIXME fails if enabled and persistence
+    }
+
+    entityProbe.expectMsgType[Started](30.seconds)
+  }
 
   def setStoreIfNotDdata(sys: ActorSystem): Unit =
     if (!isDdataMode) {

